pytorch_lightning/callbacks/progress/base.py

```diff
@@ -128,7 +147,18 @@ def total_predict_batches(self) -> Union[int, float]:
         Use this to set the total number of iterations in the progress bar. Can return ``inf`` if the predict dataloader
         is of infinite size.
         """
-        return sum(self.trainer.num_predict_batches)
+        assert self._current_eval_dataloader_idx is not None
+        return self.trainer.num_predict_batches[self._current_eval_dataloader_idx]
+
+    @property
+    def num_val_batches(self) -> Union[int, float]:
```

Comment:
Why do we need this distinction?

The name is too similar to `total_val_batches`

Reply:
yeah! for main progress bar, we need the sum of all val batches, but for the individual val progress bar, we just need the current one. We can move this to tqdm progress bar module directly, but then we would need to do the same for rich as well, so thought maybe keep a common one.

or maybe a better name for this property?

Reply:
`total_val_batches_current_epoch`?